File: framework/src/main/java/com/github/mrmitew/bodylog/framework/common/view/Presentable.java
package com.github.mrmitew.bodylog.framework.common.view;

import com.github.mrmitew.bodylog.adapter.common.view.BaseView;
import com.github.mrmitew.bodylog.framework.common.presenter.BasePresenterHolder;

public interface Presentable<V extends BaseView<S>, S> {
    V getView();

    BasePresenterHolder<V, S> injectPresenterHolder();
}

File: framework/src/main/java/com/github/mrmitew/bodylog/framework/common/view/Presentable.kt
package com.github.mrmitew.bodylog.framework.common.view

import com.github.mrmitew.bodylog.adapter.common.UiState
import com.github.mrmitew.bodylog.adapter.common.view.BaseView
import com.github.mrmitew.bodylog.framework.common.presenter.BasePresenterHolder

interface Presentable<V : BaseView<S>, S : UiState> {
    val view: V

    fun injectPresenterHolder(): BasePresenterHolder<V, S>
}
